eth_enc424j600 fails to build

When doing:

`./scripts/sanitycheck -p reel_board -s samples/net/zperf/sample.net.zperf.shield `

Get:

```
/home/galak/git/zephyr/drivers/ethernet/eth_enc424j600.c: In function 'enc424j600_init':
/home/galak/git/zephyr/drivers/ethernet/eth_enc424j600.c:603:9: error: statement with no effect [-Werror=unused-value]
  603 |  context->dev;
      |  ~~~~~~~^~~~~
cc1: all warnings being treated as errors

```

--- a/drivers/ethernet/eth_enc424j600.c
+++ b/drivers/ethernet/eth_enc424j600.c
@@ -600,8 +600,6 @@ static int enc424j600_init(const struct device *dev)
 	uint8_t retries = ENC424J600_DEFAULT_NUMOF_RETRIES;
 	uint16_t tmp;
 
-	context->dev;
-
 	/* SPI config */
 	context->spi_cfg.operation = SPI_WORD_SET(8);
 	context->spi_cfg.frequency = config->spi_freq;

